lib/compiler/src/beam_ssa_opt.erl

```diff
@@ -721,11 +817,16 @@ live_opt_phis(Is, L, Live0, LiveMap0) ->
             LiveMap;
         [_|_] ->
             PhiArgs = append([Args || #b_set{args=Args} <- Phis]),
-            PhiVars = [{P,V} || {#b_var{name=V},P} <- PhiArgs],
-            PhiLive0 = rel2fam(PhiVars),
-            PhiLive = [{{L,P},gb_sets:union(gb_sets:from_list(Vs), Live0)} ||
-                          {P,Vs} <- PhiLive0],
-            maps:merge(LiveMap, maps:from_list(PhiLive))
+            case [{P,V} || {#b_var{name=V},P} <- PhiArgs] of
```

Comment:
This is identical to a part of `beam_ssa_dead:used_vars_phis/4`, is this a general idea worth extracting?

Reply:
It is not exactly identical. This code uses gb_sets, `beam_ssa_dead` uses ordsets.